Compiler warnings in settings module in Zephyr 1.14

@nvlsianpu 

We are getting a number of Compiler Warnings when building Zephyr 1.14 (See below).
The variable rc is not initialized in these functions and could potentially get returned without being set to a value. In actual fact it may not be possible for this to happen due to the values passed into these functions being correct, but the Compiler cannot determine this, and it would be safer to actively prevent this possibility in the functions themselves.

Can you please advise on the best way to correct these warnings?
It seems that simply initializing the rc variables to an invalid value when they are declared would achieve this - do you agree?

Regards,
Declan Traill
Setec Pty Ltd

[50/254] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/settings/src/settings_store.c.obj
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_store.c: In function 'settings_cmp':
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_store.c:113:9: warning: 'rc' may be used uninitialized in this function [-Wmaybe-uninitialized]
  return rc;
         ^~
[52/254] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/settings/src/settings_line.c.obj
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_line.c: In function 'settings_line_raw_read_until':
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_line.c:272:19: warning: 'rc' may be used uninitialized in this function [-Wmaybe-uninitialized]
   return (rc) ? 0 : 1;
          ~~~~~~~~~^~~
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_line.c: In function 'settings_entry_copy':
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_line.c:444:9: warning: 'rc' may be used uninitialized in this function [-Wmaybe-uninitialized]
  return rc;
         ^~
[55/254] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/settings/src/settings_fcb.c.obj
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_fcb.c: In function 'settings_fcb_save':
C:/Users/declan.traill/RVMN101/zephyr/subsys/settings/src/settings_fcb.c:270:5: warning: 'rc' may be used uninitialized in this function [-Wmaybe-uninitialized]
  if (rc) {
     ^

--- a/subsys/settings/src/settings_fcb.c
+++ b/subsys/settings/src/settings_fcb.c
@@ -249,7 +249,7 @@ static int settings_fcb_save(struct settings_store *cs, const char *name,
 	struct settings_fcb *cf = (struct settings_fcb *)cs;
 	struct fcb_entry_ctx loc;
 	int len;
-	int rc;
+	int rc = -EINVAL;
 	int i;
 	u8_t wbs;
 

--- a/subsys/settings/src/settings_line.c
+++ b/subsys/settings/src/settings_line.c
@@ -226,7 +226,7 @@ static int settings_line_raw_read_until(off_t seek, char *out, size_t len_req,
 	size_t exp_size, read_size;
 	u8_t rbs = settings_io_cb.rwbs;
 	off_t off;
-	int rc;
+	int rc = -EINVAL;
 
 	rem_size = len_req;
 
@@ -414,7 +414,7 @@ int settings_line_name_read(char *out, size_t len_req, size_t *len_read,
 int settings_entry_copy(void *dst_ctx, off_t dst_off, void *src_ctx,
 			off_t src_off, size_t len)
 {
-	int rc;
+	int rc = -EINVAL;
 	char buf[16];
 	size_t chunk_size;
 

--- a/subsys/settings/src/settings_store.c
+++ b/subsys/settings/src/settings_store.c
@@ -86,7 +86,7 @@ static int settings_cmp(char const *val, size_t val_len, void *val_read_cb_ctx,
 	size_t len_read, exp_len;
 	size_t rem;
 	char buf[16];
-	int rc;
+	int rc = -EINVAL;
 	off_t off = 0;
 
 	for (rem = val_len; rem > 0; rem -= len_read) {

